test/integration/acceptance/queries_acceptance_test.cpp

```diff
@@ -0,0 +1,466 @@
+
+#include "backend/protobuf/transaction.hpp"
+#include "framework/integration_framework/integration_test_framework.hpp"
+#include "framework/specified_visitor.hpp"
+#include "integration/acceptance/acceptance_fixture.hpp"
+#include "interfaces/permissions.hpp"
+#include "utils/query_error_response_visitor.hpp"
+
+using namespace integration_framework;
+using namespace shared_model;
+
+class QueriesAcceptanceTest : public AcceptanceFixture {
+ public:
+  QueriesAcceptanceTest()
+      : itf(1),
+        invalidPrivateKey(kUserKeypair.privateKey().hex()),
+        invalidPublicKey(kUserKeypair.publicKey().hex()) {
+    /*
+     * It's deliberately get broken the public key and privite key to simulate a
+     * non-valid signature and public key and use their combinations in the
+     * tests below
+     * Both keys are hex values represented as a std::string so
+     * characters can be values only in the range 0-9 and a-f
+     */
+    invalidPrivateKey[0] == '9' or invalidPrivateKey[0] == 'f'
+        ? --invalidPrivateKey[0]
+        : ++invalidPrivateKey[0];
+    invalidPublicKey[0] == '9' or invalidPublicKey[0] == 'f'
+        ? --invalidPublicKey[0]
+        : ++invalidPublicKey[0];
+  };
+  void SetUp() {
+    itf.setInitialState(kAdminKeypair)
+        .sendTx(makeUserWithPerms({interface::permissions::Role::kGetRoles}))
+        .skipProposal()
+        .checkBlock([](auto &block) {
+          ASSERT_EQ(boost::size(block->transactions()), 1);
+        });
+  };
+
+  template<typename F>
+  auto checkIfStatefulFailed(F& queryResponse)
+  {
+    ASSERT_TRUE(
+        boost::apply_visitor(interface::QueryErrorResponseChecker<
+                                 interface::StatefulFailedErrorResponse>(),
+                             queryResponse.get()))
+                  << "Actual response: " << queryResponse.toString();
+  };
+
+  template<typename F>
+  auto checkIfStatelessFailed(F& queryResponse)
+  {
+    ASSERT_TRUE(
+        boost::apply_visitor(interface::QueryErrorResponseChecker<
+                                 interface::StatelessFailedErrorResponse>(),
+                             queryResponse.get()))
+                  << "Actual response: " << queryResponse.toString();
+  };
+
+
+  template<typename F>
+  auto checkIfSuccess(F& queryResponse)
+  {
+    ASSERT_NO_THROW(boost::apply_visitor(
+        framework::SpecifiedVisitor<interface::RolesResponse>(),
+        queryResponse.get()));
+
+  }
+
+  IntegrationTestFramework itf;
+  std::string invalidPrivateKey;
+  std::string invalidPublicKey;
+  const std::string NonExistentUserId = "aaaa@aaaa";
+};
+
+/**
+ * @given query with a non-existent creator_account_id
+ * @when execute any correct query with kGetRoles permissions
+ * @then the query should not pass stateful validation
+ */
+TEST_F(QueriesAcceptanceTest, NonExistentCreatorId) {
+
+  auto check = [this](auto &queryResponse){checkIfStatefulFailed(queryResponse);};
+
+  auto query = baseQry()
```

Comment:
Why have you changed TestUnsignedQueryBuilder to baseQry at the only place?

Reply:
Usage of baseQry and complete will let you reduce the amount of code to be typed. Please check thier implementations.

For example, here query creation will look like as follows:
```
auto query = complete(baseQry(NonExistentUserId).getRoles());
```
In order to use that feature `baseQry(account_id)` you have to merge the latest develop to this branch.